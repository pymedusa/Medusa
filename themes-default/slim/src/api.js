import axios from 'axios';

// This should be more dynamic. As now when we change the apiKey in config/general. This won't work anymore.
// Because of this, a page reload is required.
export const webRoot = document.body.getAttribute('web-root');
export const apiKey = document.body.getAttribute('api-key');

/**
 * Api client based on the axios client, to communicate with medusa's web routes, which return json data.
 */
export const apiRoute = axios.create({
    baseURL: webRoot + '/',
    timeout: 60000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v1.
 */
export const apiv1 = axios.create({
    baseURL: webRoot + '/api/v1/' + apiKey + '/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
    }
});

/**
 * Api client based on the axios client, to communicate with medusa's api v2.
 */
export const api = axios.create({
    baseURL: webRoot + '/api/v2/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        'X-Api-Key': apiKey
    }
});
