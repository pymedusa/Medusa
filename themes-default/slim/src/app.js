import Vue from 'vue';
import Vuex, { mapState } from 'vuex';
import VueRouter from 'vue-router';
import AsyncComputed from 'vue-async-computed';
import ToggleButton from 'vue-js-toggle-button';
import Snotify from 'vue-snotify';
import Truncate from 'vue-truncate-collapsed';
import store from './store';
import router from './router';
import { isDevelopment } from './utils';
import {
    AnidbReleaseGroupUi,
    AppHeader,
    AppLink,
    Asset,
    Backstretch,
    Config,
    DisplayShow,
    FileBrowser,
    LanguageSelect,
    NamePattern,
    PlotInfo,
    RootDirs,
    ScrollButtons,
    SelectList,
    ShowSelector
} from './components';

Vue.config.devtools = true;
Vue.config.performance = true;

Vue.use(Vuex);
Vue.use(VueRouter);
Vue.use(AsyncComputed);
Vue.use(ToggleButton);
Vue.use(Snotify);

// Load x-template components
window.components.forEach(component => {
    if (isDevelopment) {
        console.debug(`Registering ${component.name}`);
    }
    Vue.component(component.name, component);
});

// Global components
const globalComponents = [
    AnidbReleaseGroupUi,
    AppHeader,
    AppLink,
    Asset,
    Backstretch,
    Config,
    DisplayShow,
    FileBrowser,
    LanguageSelect,
    NamePattern,
    PlotInfo,
    RootDirs,
    ScrollButtons,
    SelectList,
    ShowSelector
];

globalComponents.forEach(component => {
    Vue.component(component.name, component);
});

const app = new Vue({
    name: 'App',
    store,
    router,
    components: {
        Truncate
    },
    data() {
        return {
            globalLoading: false,
            pageComponent: false
        };
    },
    computed: Object.assign(mapState(['auth', 'config']), {}),
    mounted() {
        if (isDevelopment) {
            console.log('App Mounted!');
        }

        if (!document.location.pathname.includes('/login')) {
            const { $store } = this;
            $store.dispatch('login', { username: window.username });
            $store.dispatch('getConfig');

            if (isDevelopment) {
                console.log('App Loaded!');
            }
        }
    }
}).$mount('#vue-wrap');

export default app;
