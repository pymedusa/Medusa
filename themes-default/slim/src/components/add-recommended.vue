<template>
    <div id="addShowPortal">
        <app-link href="addShows/trendingShows/?traktList=anticipated" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addtrakt"></div></div>
            <div class="buttontext">
                <h3>Add From Trakt Lists</h3>
                <p>For shows that you haven't downloaded yet, this option lets you choose from a show from one of the Trakt lists to add to Medusa .</p>
            </div>
        </app-link>

        <app-link href="addShows/popularShows/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addimdb"></div></div>
            <div class="buttontext">
                <h3>Add From IMDB's Popular Shows</h3>
                <p>View IMDB's list of the most popular shows. This feature uses IMDB's MOVIEMeter algorithm to identify popular TV Shows.</p>
            </div>
        </app-link>

        <app-link href="addShows/popularAnime/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addanime"></div></div>
            <div class="buttontext">
                <h3>Add From Anidb's Hot Anime list</h3>
                <p>View Anidb's list of the most popular anime shows. Anidb provides lists for Popular Anime, using the "Hot Anime" list.</p>
            </div>
        </app-link>
    </div>
</template>
<script>
import AppLink from './app-link.vue';

export default {
    name: 'addRecommended',
    components: {
        AppLink
    },
    computed: {
        header() {
            return this.$route.meta.header;
        }
    }
};
</script>
<style>
/* placeholder */
</style>
