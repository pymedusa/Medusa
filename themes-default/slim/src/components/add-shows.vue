<template>
    <div id="addShowPortal">
        <app-link href="addShows/newShow/" id="btnNewShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addnewshow"></div></div>
            <div class="buttontext">
                <h3>Add New Show</h3>
                <p>For shows that you haven't downloaded yet, this option finds a show on your preferred indexer, creates a directory for it's episodes, and adds it to Medusa.</p>
            </div>
        </app-link>

        <app-link href="addShows/existingShows/" id="btnExistingShow" class="btn-medusa btn-large">
            <div class="button"><div class="add-list-icon-addexistingshow"></div></div>
            <div class="buttontext">
                <h3>Add Existing Shows</h3>
                <p>Use this option to add shows that already have a folder created on your hard drive. Medusa will scan your existing metadata/episodes and add the show accordingly.</p>
            </div>
        </app-link>
    </div>
</template>
<script>
import AppLink from './app-link.vue';

export default {
    name: 'addShows',
    components: {
        AppLink
    },
    computed: {
        header() {
            return this.$route.meta.header;
        }
    }
};
</script>
<style>
/* placeholder */
</style>
