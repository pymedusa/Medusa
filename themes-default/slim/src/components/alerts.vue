<template>
    <div v-if="system.branch !== null" id="alerts">
        <!-- BEGIN ALERTS -->
        <div v-if="system.branch !== 'master' && !developer" class="text-center">
            <div class="alert alert-danger upgrade-notification hidden-print" role="alert">
                <span>You're using the {{system.branch}} branch. Please use 'master' unless specifically asked</span>
            </div>
        </div>

        <div v-if="system.newestVersionMessage" class="text-center">
            <div class="alert alert-success upgrade-notification hidden-print" role="alert">
                <span v-html="system.newestVersionMessage" />
            </div>
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';
export default {
    name: 'alerts',
    computed: {
        ...mapState({
            system: state => state.config.system,
            developer: state => state.config.general.developer
        })
    }
};
</script>

<style>
</style>
