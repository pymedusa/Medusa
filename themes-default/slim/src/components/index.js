export { default as AddRecommended } from './add-recommended.vue';
export { default as AddShowOptions } from './add-show-options.vue';
export { default as AddShows } from './add-shows.vue';
export { default as AnidbReleaseGroupUi } from './anidb-release-group-ui.vue';
export { default as AppFooter } from './app-footer.vue';
export { default as AppHeader } from './app-header.vue';
export { default as Backstretch } from './backstretch.vue';
export { default as Config } from './config.vue';
export { default as ConfigGeneral } from './config-general.vue';
export { default as ConfigPostProcessing } from './config-post-processing.vue';
export { default as ConfigNotifications } from './config-notifications.vue';
export { default as ConfigSearch } from './config-search.vue';
export { default as DisplayShow } from './display-show.vue';
export { default as EditShow } from './edit-show.vue';
export { default as History } from './history.vue';
export { default as Home } from './home.vue';
export { default as IRC } from './irc.vue';
export { default as Login } from './login.vue';
export { default as Logs } from './logs.vue';
export { default as ManualPostProcess } from './manual-post-process.vue';
export { default as RootDirs } from './root-dirs.vue';
export { default as Schedule } from './schedule.vue';
export { default as ShowHeader } from './show-header.vue';
export { default as ShowHistory } from './show-history.vue';
export { default as ShowResults } from './show-results.vue';
export { default as SnatchSelection } from './snatch-selection.vue';
export { default as Status } from './status.vue';
export { default as SubMenu } from './sub-menu.vue';
export { default as SubtitleSearch } from './subtitle-search.vue';
export * from './http';
export * from './helpers';
