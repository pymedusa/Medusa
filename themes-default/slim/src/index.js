import $ from 'jquery';
import 'bootstrap'; // eslint-disable-line import/no-unassigned-import
import 'bootstrap/dist/css/bootstrap.min.css'; // eslint-disable-line import/no-unassigned-import
import './css/open-sans.css'; // eslint-disable-line import/no-unassigned-import

import Vue from 'vue';
import Vuex from 'vuex';
import VueMeta from 'vue-meta';
import VueRouter from 'vue-router';
import VueNativeSock from 'vue-native-websocket';
import AsyncComputed from 'vue-async-computed';
import ToggleButton from 'vue-js-toggle-button';
import Snotify from 'vue-snotify';
import axios from 'axios';
import debounce from 'lodash/debounce';
import store from './store';
import router from './router';
import { isDevelopment } from './utils';
import { apiRoute, apiv1, api, webRoot, apiKey } from './api';
import { Asset, Backstretch, DisplayShow, PlotInfo, ShowSelector, ScrollButtons, FileBrowser, NamePattern, AppHeader, AppLink, SelectList, LanguageSelect, RootDirs, AnidbReleaseGroupUi, SnatchSelection } from './components';

if (window) {
    window.isDevelopment = isDevelopment;

    // Adding libs to window so mako files can use them
    window.$ = $;
    window.jQuery = $;
    window.Vue = Vue;
    window.Vuex = Vuex;
    window.VueMeta = VueMeta;
    window.VueRouter = VueRouter;
    window.VueNativeSock = VueNativeSock;
    window.AsyncComputed = AsyncComputed;
    window.ToggleButton = ToggleButton;
    window.Snotify = Snotify;
    window.axios = axios;
    window._ = { debounce };
    window.store = store;
    window.router = router;
    window.apiRoute = apiRoute;
    window.apiv1 = apiv1;
    window.api = api;

    window.MEDUSA = {
        common: {},
        config: {},
        home: {},
        manage: {},
        history: {},
        errorlogs: {},
        schedule: {},
        addShows: {}
    };
    window.webRoot = webRoot;
    window.apiKey = apiKey;
    window.apiRoot = webRoot + '/api/v2/';

    // Push pages that load via a vue file but still use `el` for mounting
    window.components = [];
    window.components.push(AnidbReleaseGroupUi);
    window.components.push(AppHeader);
    window.components.push(AppLink);
    window.components.push(Asset);
    window.components.push(Backstretch);
    window.components.push(DisplayShow);
    window.components.push(FileBrowser);
    window.components.push(LanguageSelect);
    window.components.push(NamePattern);
    window.components.push(PlotInfo);
    window.components.push(RootDirs);
    window.components.push(ScrollButtons);
    window.components.push(SelectList);
    window.components.push(ShowSelector);
    window.components.push(SnatchSelection);
}
const UTIL = {
    exec(controller, action) {
        const ns = MEDUSA;
        action = (action === undefined) ? 'init' : action;

        if (controller !== '' && ns[controller] && typeof ns[controller][action] === 'function') {
            ns[controller][action]();
        }
    },
    init() {
        if (typeof startVue === 'function') { // eslint-disable-line no-undef
            startVue(); // eslint-disable-line no-undef
        } else {
            $('[v-cloak]').removeAttr('v-cloak');
        }

        const { body } = document;
        $('[asset]').each(function() {
            const asset = $(this).attr('asset');
            const series = $(this).attr('series');
            const path = apiRoot + 'series/' + series + '/asset/' + asset + '?api_key=' + apiKey;
            if (this.tagName.toLowerCase() === 'img') {
                const defaultPath = $(this).attr('src');
                if ($(this).attr('lazy') === 'on') {
                    $(this).attr('data-original', path);
                } else {
                    $(this).attr('src', path);
                }
                $(this).attr('onerror', 'this.src = "' + defaultPath + '"; return false;');
            }
            if (this.tagName.toLowerCase() === 'a') {
                $(this).attr('href', path);
            }
        });
        const controller = body.getAttribute('data-controller');
        const action = body.getAttribute('data-action');

        UTIL.exec('common'); // Load common
        UTIL.exec(controller); // Load MEDUSA[controller]
        UTIL.exec(controller, action); // Load MEDUSA[controller][action]

        window.dispatchEvent(new Event('medusa-loaded'));
    }
};

$.fn.extend({
    addRemoveWarningClass(_) {
        if (_) {
            return $(this).removeClass('warning');
        }
        return $(this).addClass('warning');
    }
});

const { pathname } = window.location;
if (!pathname.includes('/login') && !pathname.includes('/apibuilder')) {
    api.get('config/main').then(response => {
        $.extend(MEDUSA.config, response.data);
        MEDUSA.config.themeSpinner = MEDUSA.config.themeName === 'dark' ? '-dark' : '';
        MEDUSA.config.loading = '<img src="images/loading16' + MEDUSA.config.themeSpinner + '.gif" height="16" width="16" />';

        if (navigator.userAgent.indexOf('PhantomJS') === -1) {
            $(document).ready(UTIL.init);
        }

        MEDUSA.config.indexers.indexerIdToName = indexerId => {
            if (!indexerId) {
                return '';
            }
            return Object.keys(MEDUSA.config.indexers.config.indexers).filter(indexer => { // eslint-disable-line array-callback-return
                if (MEDUSA.config.indexers.config.indexers[indexer].id === parseInt(indexerId, 10)) {
                    return MEDUSA.config.indexers.config.indexers[indexer].name;
                }
            })[0];
        };

        MEDUSA.config.indexers.nameToIndexerId = name => {
            if (!name) {
                return '';
            }
            return MEDUSA.config.indexers.config.indexers[name];
        };
    }).catch(error => {
        console.debug(error);
        alert('Unable to connect to Medusa!'); // eslint-disable-line no-alert
    });
}
