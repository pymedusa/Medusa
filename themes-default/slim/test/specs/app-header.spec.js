import Vuex, { Store } from 'vuex';
import VueRouter from 'vue-router';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import { AppHeader } from '../../src/components';
import fixtures from '../__fixtures__/common';

describe('AppHeader.test.js', () => {
    let localVue;
    let store;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuex);
        localVue.use(VueRouter);

        const { state } = fixtures;
        store = new Store({ state });
    });

    it('renders', () => {
        const { state } = fixtures;
        const wrapper = shallowMount(AppHeader, {
            localVue,
            store,
            computed: {
                config() {
                    return {
                        ...state.config
                    };
                },
                topMenu() {
                    return 'home';
                }
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
