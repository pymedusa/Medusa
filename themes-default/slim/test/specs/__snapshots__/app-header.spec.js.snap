// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.test.js renders 1`] = `
<nav
  class="navbar navbar-default navbar-fixed-top hidden-print"
  role="navigation"
>
  <div
    class="container-fluid"
  >
    <div
      class="navbar-header"
    >
      <button
        class="navbar-toggle collapsed"
        data-target="#main_nav"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="floating-badge"
        >
          1
        </span>
         
        <span
          class="sr-only"
        >
          Toggle navigation
        </span>
         
        <span
          class="icon-bar"
        />
         
        <span
          class="icon-bar"
        />
         
        <span
          class="icon-bar"
        />
      </button>
       
      <app-link-stub
        class="navbar-brand"
        href="home/"
        placeholder="indexer-to-name"
        title="Medusa"
      >
        <img
          alt="Medusa"
          class="img-responsive pull-left"
          src="images/medusa.png"
          style="height: 50px;"
        />
      </app-link-stub>
    </div>
     
    <div
      class="collapse navbar-collapse"
      id="main_nav"
    >
      <ul
        class="nav navbar-nav navbar-right"
      >
        <li
          class="navbar-split dropdown active"
          id="NAVhome"
        >
          <app-link-stub
            aria-haspopup="true"
            class="dropdown-toggle"
            data-hover="dropdown"
            data-toggle="dropdown"
            href="home/"
            placeholder="indexer-to-name"
          >
            <span>
              Shows
            </span>
             
            <b
              class="caret"
            />
          </app-link-stub>
           
          <ul
            class="dropdown-menu"
          >
            <li>
              <app-link-stub
                href="home/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-home"
                />
                 Show List
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="addShows/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-addshow"
                />
                 Add Shows
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="addRecommended/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-addshow"
                />
                 Add Recommended Shows
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="home/postprocess/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-postprocess"
                />
                 Manual Post-Processing
              </app-link-stub>
            </li>
             
            <li
              class="divider"
              role="separator"
            />
             
            <li>
              <app-link-stub
                href="home/displayShow?indexername=tvdb&seriesid=253463"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-addshow"
                />
                 Black Mirror
                            
              </app-link-stub>
            </li>
            <li>
              <app-link-stub
                href="home/displayShow?indexername=tvdb&seriesid=80379"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-addshow"
                />
                 The Big Bang Theory
                            
              </app-link-stub>
            </li>
          </ul>
           
          <div
            style="clear: both;"
          />
        </li>
         
        <li
          class="navbar-split"
          id="NAVschedule"
        >
          <app-link-stub
            href="schedule/"
            placeholder="indexer-to-name"
          >
            Schedule
          </app-link-stub>
        </li>
         
        <li
          class="navbar-split"
          id="NAVhistory"
        >
          <app-link-stub
            href="history/"
            placeholder="indexer-to-name"
          >
            History
          </app-link-stub>
        </li>
         
        <li
          class="navbar-split dropdown"
          id="NAVmanage"
        >
          <app-link-stub
            aria-haspopup="true"
            class="dropdown-toggle"
            data-hover="dropdown"
            data-toggle="dropdown"
            href="manage/episodeStatuses/"
            placeholder="indexer-to-name"
          >
            <span>
              Manage
            </span>
             
            <b
              class="caret"
            />
          </app-link-stub>
           
          <ul
            class="dropdown-menu"
          >
            <li>
              <app-link-stub
                href="manage/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-manage"
                />
                 Mass Update
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="manage/backlogOverview/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-backlog-view"
                />
                 Backlog Overview
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="manage/manageSearches/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-manage-searches"
                />
                 Manage Searches
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="manage/episodeStatuses/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-manage2"
                />
                 Episode Status Management
              </app-link-stub>
            </li>
             
            <!---->
             
            <li>
              <app-link-stub
                href="home/updateKODI/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-kodi"
                />
                 Update KODI
              </app-link-stub>
            </li>
             
            <!---->
             
            <!---->
             
            <li>
              <app-link-stub
                href="manage/failedDownloads/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-failed-download"
                />
                 Failed Downloads
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="manage/subtitleMissed/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-backlog"
                />
                 Missed Subtitle Management
              </app-link-stub>
            </li>
             
            <!---->
          </ul>
           
          <div
            style="clear: both;"
          />
        </li>
         
        <li
          class="navbar-split dropdown"
          id="NAVconfig"
        >
          <app-link-stub
            aria-haspopup="true"
            class="dropdown-toggle"
            data-hover="dropdown"
            data-toggle="dropdown"
            href="config/"
            placeholder="indexer-to-name"
          >
            <span
              class="visible-xs-inline"
            >
              Config
            </span>
            <img
              class="navbaricon hidden-xs"
              src="images/menu/system18.png"
            />
             
            <b
              class="caret"
            />
          </app-link-stub>
           
          <ul
            class="dropdown-menu"
          >
            <li>
              <app-link-stub
                href="config/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-help"
                />
                 Help & Info
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/general/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-config"
                />
                 General
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/backuprestore/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-backup"
                />
                 Backup & Restore
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/search/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-manage-searches"
                />
                 Search Settings
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/providers/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-provider"
                />
                 Search Providers
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/subtitles/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-backlog"
                />
                 Subtitles Settings
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/postProcessing/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-postprocess"
                />
                 Post-Processing
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/notifications/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-notification"
                />
                 Notifications
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="config/anime/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-anime"
                />
                 Anime
              </app-link-stub>
            </li>
          </ul>
           
          <div
            style="clear: both;"
          />
        </li>
         
        <li
          class="navbar-split dropdown"
          id="NAVsystem"
        >
          <app-link-stub
            aria-haspopup="true"
            class="padding-right-15 dropdown-toggle"
            data-hover="dropdown"
            data-toggle="dropdown"
            href="home/status/"
            placeholder="indexer-to-name"
          >
            <span
              class="visible-xs-inline"
            >
              Tools
            </span>
            <img
              class="navbaricon hidden-xs"
              src="images/menu/system18-2.png"
            />
             
            <span
              class="badge"
            >
              1
            </span>
             
            <b
              class="caret"
            />
          </app-link-stub>
           
          <ul
            class="dropdown-menu"
          >
            <li>
              <app-link-stub
                href="news/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-news"
                />
                 News 
                <span
                  class="badge"
                >
                  1
                </span>
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="IRC/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-irc"
                />
                 IRC
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="changes/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-changelog"
                />
                 Changelog
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="https://github.com/pymedusa/Medusa/wiki/Donations"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-support"
                />
                 Support Medusa
              </app-link-stub>
            </li>
             
            <li
              class="divider"
              role="separator"
            />
             
            <!---->
             
            <!---->
             
            <li>
              <app-link-stub
                href="errorlogs/viewlog/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-viewlog"
                />
                 View Log
              </app-link-stub>
            </li>
             
            <li
              class="divider"
              role="separator"
            />
             
            <li>
              <app-link-stub
                href="home/updateCheck?pid=1234"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-update"
                />
                 Check For Updates
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="home/restart/?pid=1234"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-restart"
                />
                 Restart
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="home/shutdown/?pid=1234"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-shutdown"
                />
                 Shutdown
              </app-link-stub>
            </li>
             
            <li>
              <app-link-stub
                href="logout"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-shutdown"
                />
                 Logout
              </app-link-stub>
            </li>
             
            <li
              class="divider"
              role="separator"
            />
             
            <li>
              <app-link-stub
                href="home/status/"
                placeholder="indexer-to-name"
              >
                <i
                  class="menu-icon-info"
                />
                 Server Status
              </app-link-stub>
            </li>
          </ul>
           
          <div
            style="clear: both;"
          />
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
